core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -322,16 +322,6 @@ abstract class BaseColumnFormatRelation(
 
       if (tableExists) {
         mode match {
-          case SaveMode.Ignore =>
-            dialect match {
-              case GemFireXDDialect =>
-                GemFireXDDialect.initializeTable(table,
-                  sqlContext.conf.caseSensitiveAnalysis, conn)
-                GemFireXDDialect.initializeTable(externalColumnTableName,
-                  sqlContext.conf.caseSensitiveAnalysis, conn)
-              case _ => // Do nothing
-            }
-            return
           case SaveMode.ErrorIfExists =>
             // sys.error(s"Table $table already exists.") TODO: Why so?
             return
```

Comment:
Perhaps we would want to move the initializeTable in the tableExists==false case i.e. when table is just being created in store. The case could be the thin connector mode where having bucket information would be better for routing else need to take care that first few routing may be incomplete/incorrect (no caching of bucket information, if any, etc). If missing bucket information is handled cleanly in thin connector then above would be just fine.